### PR TITLE
Add python3-streamz to rosdep/python.yaml

### PR DESCRIPTION
**Pull request by galou**
Please add the following dependency to the rosdep database.

## Package name:

`python3-streamz`: Pipelines to manage continuous streams of data

## Package Upstream Source:

https://github.com/python-streamz/streamz/

## Purpose of using this:

We use `streamz` to apply a sliding window average on raw input data for logging and user interface.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/search?keywords=streamz&searchon=names&suite=stable&section=all
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/oracular/python3-streamz
- Fedora: https://packages.fedoraproject.org/
  - not available
- Arch: https://www.archlinux.org/packages/
  - not available
- Gentoo: https://packages.gentoo.org/
  - not available
- macOS: https://formulae.brew.sh/
  - not available
- Alpine: https://pkgs.alpinelinux.org/packages
  - not available
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/development/python-modules/streamz/default.nix#L83
- openSUSE: https://software.opensuse.org/package/
  - https://software.opensuse.org/package/python3-streamz
- rhel: https://rhel.pkgs.org/
  - https://debian.pkgs.org/12/debian-main-amd64/python3-streamz_0.6.4-1_all.deb.html

### CODE REVIEW

**Review by sloretz — DISMISSED** [review]
It looks like this package is available as system packages. Please use those instead of `pip` entries.

```
Key 'python3-streamz' might be satisfied by debian package named 'python3-streamz': https://packages.debian.org/bookworm/python3-streamz
```

```
Key 'python3-streamz' might be satisfied by opensuse package named 'python3-streamz': https://software.opensuse.org/package/python-streamz
```

**Comment by galou** [comment]
There's also the package for Tumbleweed but I don't know how to add it (`opensuse: [python-streamz]`?).

**Comment by galou** [comment]
What is preferred, should I rebase and squash or simply push a new commit?

**Comment by clalancette** [comment]
> What is preferred, should I rebase and squash or simply push a new commit?

Either one is fine.  We'll end up squashing it in the end.

File: rosdep/python.yaml [hunk 1]
```diff
@@ -9886,6 +9886,18 @@ python3-streamlit-pip:
...
       packages: [streamlit]
+python3-streamz:
+  debian:
+    bookworm: [python3-streamz]
+    bullseye: [python3-streamz]
+  ubuntu:
+    '*': [python3-streamz]
+    bionic:
+      pip:
+        packages: [streamz]
```
Comment (clalancette) [inline rosdep/python.yaml:9897]: Bionic is EOL, so we don't need to add it:

```suggestion
```

File: rosdep/python.yaml [hunk 2]
```diff
@@ -9886,6 +9886,18 @@ python3-streamlit-pip:
   '*':
     pip:
       packages: [streamlit]
+python3-streamz:
+  debian:
+    bookworm: [python3-streamz]
+    bullseye: [python3-streamz]
+  ubuntu:
```
Comment (clalancette) [inline rosdep/python.yaml:9893]:
```suggestion
  opensuse: [python3-streamz]
  ubuntu:
```

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/python.yaml (hunk 2 above)
Comment (galou) [reply rosdep/python.yaml:9893]: `python3-streamz` is only available as experimental package if I understand correctly. Moreover, on Tumbleweed it is available as `python-streamz`.

**Comment by galou** [comment]
I committed the suggested changes because I got no answer about my doubts with OpenSuse.

**Review by clalancette — APPROVED** [review]

**Review by quarkytale — APPROVED** [review]

**Pull request merged**